File: atcoder.jp/PyPy3 (7.3.0)/abc250/abc250_d.py
N = int(input())

def prime(n):
    is_prime = [True] * (n+1)
    is_prime[0] = is_prime[1] = False
    for p in range(2,int(n**0.5)+1):
        if is_prime[p]:
            for i in range(p*p, n+1, p):
                is_prime[i] = False
    return [i for i in range(n+1) if is_prime[i]]

limit = int(pow(N, 1/3))+1
primes = prime(limit)

cnt = 0
length = len(primes)
for i in range(length-1):
    ok = i
    ng = length
    while ng - ok > 1:
        mid = (ok + ng) // 2
        if primes[mid] ** 3 * primes[i] <= N:
            ok = mid
        else:
            ng = mid
    cnt += ok - i

print(cnt)

File: atcoder.jp/Rust (1.42.0)/abc250/abc250_d.rs
use proconio::{fastout, input};

fn prime(n: usize) -> Vec<usize> {
    let mut prime = vec![true; n + 1];
    prime[0] = false;
    prime[1] = false;
    for i in 2..(n + 1) {
        if prime[i] {
            for j in (i * i..(n + 1)).step_by(i) {
                prime[j] = false;
            }
        }
    }
    prime
        .iter()
        .enumerate()
        .filter(|&(_, &p)| p)
        .map(|(i, _)| i)
        .collect()
}

const LIMIT: usize = 1000000;

#[fastout]
fn main() {
    let primes = prime(LIMIT);
    let mut ans = 0;
    input! {
        n: usize,
    };

    for i in 1..primes.len() {
        let a = primes[i].pow(3);
        if a * 2 > n {
            break;
        }
        let mut ok: usize = 0;
        let mut ng: usize = i;
        while (ng - ok) > 1 {
            let mid = (ok + ng) / 2;
            if a * primes[mid] <= n {
                ok = mid;
            } else {
                ng = mid;
            }
        };
        ans += ok + 1;
    }
    println!("{}", ans);
}
